### PR TITLE
develop-jpa-repo

### PR DESCRIPTION
**Pull request by alwayswannajava**

### CODE REVIEW

File: src/main/java/com/spring/onlinebookstore/service/BookServiceImpl.java [hunk 1]
```diff
@@ -30,9 +31,47 @@ public List<BookDto> findAll() {
...
+        Book book = bookRepository.findById(id).orElseThrow(() ->
+                new EntityNotFoundException("Can't find book by id: " + id));
+        bookRepository.deleteById(book.getId());
+    }
+
+    @Override
+    public BookDto update(long id, UpdateBookRequestDto updateBookRequestDto) {
+        Book book = bookRepository.findById(id).orElseThrow(() ->
+                new EntityNotFoundException("Can't find book by id: " + id));
+        update(book, updateBookRequestDto);
```
Comment (Elena-Bruyako) [inline src/main/java/com/spring/onlinebookstore/service/BookServiceImpl.java]: use MapStruct to update book

File: src/main/java/com/spring/onlinebookstore/service/BookServiceImpl.java [hunk 2]
```diff
@@ -30,9 +31,47 @@ public List<BookDto> findAll() {
...
+        return bookMapper.toDto(book);
+    }
+
+    private void update(Book book, UpdateBookRequestDto updateBookRequestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/spring/onlinebookstore/service/BookServiceImpl.java]: remove method

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/spring/onlinebookstore/model/Book.java [hunk 3]
```diff
@@ -26,6 +30,10 @@ public class Book {
     private String isbn;
     @Column(nullable = false)
     private BigDecimal price;
+
+    @Column(columnDefinition = "tinyint(1) default 0", nullable = false)
+    private boolean isDeleted = false;
```
Comment (MentorMateAcademy) [inline src/main/java/com/spring/onlinebookstore/model/Book.java]: `@Column(nullable = false)
@Type(type = "org.hibernate.type.NumericBooleanType")
private boolean isDeleted = false;
`
try to use this way

**Review by MentorMateAcademy — APPROVED** [review]

**Pull request merged**